typescript/cli/src/deploy/core.ts

```diff
@@ -115,29 +153,72 @@ export async function runCoreDeploy(params: DeployParams) {
 
 export async function runCoreApply(params: ApplyParams) {
   const { context, chain, deployedCoreAddresses, config } = params;
-  const { multiProvider } = context;
-  const evmCoreModule = new EvmCoreModule(multiProvider, {
-    chain,
-    config,
-    addresses: deployedCoreAddresses,
-  });
-
-  const transactions = await evmCoreModule.update(config);
-
-  if (transactions.length) {
-    logGray('Updating deployed core contracts');
-    for (const transaction of transactions) {
-      await multiProvider.sendTransaction(
-        // Using the provided chain id because there might be remote chain transactions included in the batch
-        transaction.chainId ?? chain,
-        transaction,
-      );
+  const { multiProvider, multiProtocolSigner } = context;
+
+  switch (multiProvider.getProtocol(chain)) {
```

Comment:
is it possible to dedupe more of this? part of the idea with having the abstract module is that you can somewhat initialise them and then it should just be the same function call to get update txs for each VM, and then afterwards we can iterate through and apply the txs

Reply:
I think it should be possible but it would be some more changes. I would suggest creating a single additional PR just for refactoring and trying to get rid of those (kinda) duplicated lines.